artifacts_test.go

```diff
@@ -403,3 +603,111 @@ func TestMissingArtifactFailsTest(t *testing.T) {
 
 	_ = submitAndAssert(t, td, payload, "failed", "failed")
 }
+
+func TestInvalidContentEncoding(t *testing.T) {
+
+	defer setup(t)()
+
+	expires := tcclient.Time(time.Now().Add(time.Minute * 30))
+
+	command := helloGoodbye()
+
+	payload := GenericWorkerPayload{
+		Command:	command,
+		MaxRunTime:	30,
+		Artifacts:	[]Artifact{
+			{
+				Path:			"SampleArtifacts/_/X.txt",
+				Expires:		expires,
+				Type:			"file",
+				Name:			"public/_/X.txt",
+				ContentType:		"text/plain; charset=utf-8",
+				ContentEncoding:	"jpg",
+			},
+		},
+	}
+	td := testTask(t)
+
+	_ = submitAndAssert(t, td, payload, "exception", "malformed-payload")
+
+	// check log mentions contentEncoding invalid
+	bytes, err := ioutil.ReadFile(filepath.Join(taskContext.TaskDir, logPath))
+	if err != nil {
+		t.Fatalf("Error when trying to read log file: %v", err)
+	}
+	logtext := string(bytes)
+	if !strings.Contains(logtext, "[taskcluster:error] - artifacts.0.contentEncoding: artifacts.0.contentEncoding must be one of the following: \"identity\", \"gzip\"") {
+		t.Fatalf("Was expecting log file to explain that contentEncoding was invalid, but it doesn't: \n%v", logtext)
+	}
+}
+
+func TestInvalidContentEncodingBlacklisted(t *testing.T) {
+
+	defer setup(t)()
+
+	expires := tcclient.Time(time.Now().Add(time.Minute * 30))
+
+	command := helloGoodbye()
+
+	payload := GenericWorkerPayload{
+		Command:	command,
+		MaxRunTime:	30,
+		Artifacts:	[]Artifact{
+			{
+				Path:			"SampleArtifacts/b/c/d.jpg",
+				Expires:		expires,
+				Type:			"file",
+				Name:			"public/b/c/d.jpg",
+				ContentType:		"image/jpeg",
+				ContentEncoding:	"jpg",
+			},
+		},
+	}
+	td := testTask(t)
+
+	_ = submitAndAssert(t, td, payload, "exception", "malformed-payload")
+
+	// check log mentions contentEncoding invalid
+	bytes, err := ioutil.ReadFile(filepath.Join(taskContext.TaskDir, logPath))
+	if err != nil {
+		t.Fatalf("Error when trying to read log file: %v", err)
+	}
+	logtext := string(bytes)
+	if !strings.Contains(logtext, "[taskcluster:error] - artifacts.0.contentEncoding: artifacts.0.contentEncoding must be one of the following: \"identity\", \"gzip\"") {
+		t.Fatalf("Was expecting log file to explain that contentEncoding was invalid, but it doesn't: \n%v", logtext)
+	}
+}
+
+func TestMissingContentEncoding(t *testing.T){
```

Comment:
Maybe we should call this `TestEmptyContentEncoding` since the content encoding is present.